### PR TITLE
feat: added `mainnet-fork` to mainnet providers

### PR DESCRIPTION
**Pull request by sabotagebeats**
### What I did
Add support for `mainnet-fork` to list of mainnet providers

fixes: #13 

### How I did it
added `mainnet-fork` to list of providers which are accepted mainnet providers

### How to verify it
I need to build tests for this

### Checklist

- [x] Passes all linting checks (pre-commit and CI jobs)
- [x] New test cases have been added and are passing
- [ ] Documentation has been updated
- [x] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: ape_ens/converter.py [hunk 1]
```diff
@@ -16,7 +16,12 @@ class ENSConversions(ConverterAPI):
     @cached_property
     def mainnet_provider(self) -> Optional[Web3Provider]:
         provider = self.network_manager.active_provider
-        if provider and isinstance(provider, Web3Provider) and provider.network.name == "mainnet":
+        if (
+            provider
+            and isinstance(provider, Web3Provider)
+            and provider.network.name == ("mainnet" or "mainnet-fork")
```
[the remark this answers is not in the document]
Comment (sabotagebeats) [reply ape_ens/converter.py:22]: this doesn't seem to work with `ethereum:mainnet-fork`
```
In [1]: from ape.types import AddressType

In [2]: convert("vitalik.eth", AddressType)
WARNING: Connecting Geth plugin to non-Geth client 'HardhatNetwork'.
WARNING: Unable to connect to mainnet provider to perform ENS lookup. Try changing your default mainnet provider.
```

[the remark this answers is not in the document]
Comment (sabotagebeats) [reply ape_ens/converter.py:22]: it still does work on `ethereum:mainnet:infura`

File: ape_ens/converter.py [hunk 2]
```diff
@@ -16,7 +16,12 @@ class ENSConversions(ConverterAPI):
...
+            and provider.network.ecosystem.name == "ethereum"
```
[the remark this answers is not in the document]
Comment (sabotagebeats) [reply ape_ens/converter.py:23]: my `ape-config.yaml`:
```
hardhat:
  fork:
    ethereum:
      mainnet:
        upstream_provider: alchemy
```

File: ape_ens/converter.py (hunk 1 above)
Comment (unparalleled-js) [inline ape_ens/converter.py:22]:
```suggestion
            and provider.network.name in ("mainnet", "mainnet-fork")
```

**Review by sabotagebeats — COMMENTED** [review]
👍

**Review by unparalleled-js — APPROVED** [review]

**Pull request merged**